gateway/webhook/webhook.go

```diff
@@ -124,42 +113,12 @@ func (webhook *HandleT) failRequest(w http.ResponseWriter, r *http.Request, reas
 }
 
 func (webhook *HandleT) RequestHandler(w http.ResponseWriter, r *http.Request) {
+	reqType := r.Context().Value(gwtypes.CtxParamCallType).(string)
```

Comment:
I am bit concerned with the usage of context values. Because of their impact in readability and losing compile time checks.

I need to dice deeper into the code to understand if we are able to avoid them and at what costs.

However, if we keep them I would suggest creating a function that takes a request as parameter and returns reqType.

Reply:
If it helps, all code paths which are using context values are covered with tests and if improper usage was introduced casting errors would be recorded.